### ISSUE TITLE
Load Sounds From Filesystem

### ISSUE THREAD

**Issue by BrentDaMage** [ReMinecraftPE/mcpe#119]
Currently sounds are just embedded in memory. We should update our audio API abstraction layers to support loading sounds from files. Streaming sounds from files would also allow us to add in-game music.

This means updating:
1. SoundSystemAL
2. SoundSystemDS
3. SoundSystemSL

**Comment by TheBrokenRail** [comment]
This shouldn't be that difficult. Currently sounds are stored as PCM data, which if I understand correctly, is one header away from being a WAV file. All that should be needed is to change `grabsounds.py` to extract files (with the correct header) and to make the `SoundSystem*` classes do an `fopen` and `fread` instead of just using a variable.

On the topic of `grabsounds.py` changes, I'd also suggest changing it so that it actually parses the symbol table of `libminecraftpe.so` instead of hard-coding addresses so it can use more than one APK version.

**Comment by iProgramMC** [comment]
Correct, a WAV file is pretty just PCM data with a header. It probably isn't that hard to make `grabsounds.py` output WAV files instead.

It is probably not that hard to parse the symbol table of the Minecraft PE shared object, but I haven't really felt like doing it, because to get the best experience you'll need to use the 0.1.3 assets anyway.

**Comment by BrentDaMage** [comment]
> On the topic of `grabsounds.py` changes, I'd also suggest changing it so that it actually parses the symbol table of `libminecraftpe.so` instead of hard-coding addresses so it can use more than one APK version.

On a side note, the PCM sound files can be located in the files of iOS builds as well, particularly 0.10.0.

**Comment by BrentDaMage** [comment]
We'll need OGG support for streaming music in a compressed manner.
Implementation for loading Ogg Vorbis on OpenAL: https://gist.github.com/tilkinsc/f91d2a74cff62cc3760a7c9291290b29
Implementation for loading Ogg Vorbis on DirectSound: https://www.flipcode.com/archives/Ogg_Vorbis_Player_Class.shtml
libvorbis: https://xiph.org/vorbis/
